### DISCUSSION TITLE
Allow Dmarc compliant FROM header for journal level envelope sender in multi tenent installations

### DISCUSSION THREAD

**Discussion by ronste** [pkp/pkp-lib#10673] [Proposals]
This discussion applys to all OJS 3.x versions.

Currently the settings in config.inc.php to comply with Dmarc requirements are restricted to the 'default_envelope_sender'. See relevant code (main branch):

- https://github.com/pkp/pkp-lib/blob/01210c27a04047ce68313c54ff4188b560f07fb6/classes/mail/Mailer.php#L215
- https://github.com/pkp/pkp-lib/blob/01210c27a04047ce68313c54ff4188b560f07fb6/classes/mail/Mailer.php#L236

Due to this restriction multi tenant installations using journal level baseUrls (and mail systems) cannot send Dmarc compliant mails. This restriction should be lifted and a journal level configuration of Dmarc settings should be introduced.

As currently done for the base urls the config setting could be extended to allow this.

In addition a new pattern `'%j'` should be introduced to allow the journal title to be used in the `'dmarc_compliant_from_displayname'`.

A broadly related topic and discussion already exists: https://github.com/pkp/pkp-lib/issues/6779 

Question: Are there any technical reasons I missed for imposing this restriction?

**Comment by tib-rijo** [comment]
This could also work well together with ARC signing of transactional mail.
